zephyr/CMakeLists.txt

```diff
@@ -551,19 +551,17 @@ zephyr_library_sources_ifdef(CONFIG_COMP_KPB
 	${SOF_AUDIO_PATH}/kpb.c
 )
 
-if(CONFIG_IPC_MAJOR_3)
-	zephyr_library_sources_ifdef(CONFIG_COMP_MIXER
-		${SOF_AUDIO_PATH}/mixer/mixer.c
-		${SOF_AUDIO_PATH}/mixer/mixer_generic.c
-		${SOF_AUDIO_PATH}/mixer/mixer_hifi3.c
-	)
-elseif(CONFIG_IPC_MAJOR_4)
-	zephyr_library_sources_ifdef(CONFIG_COMP_MIXER
-		${SOF_AUDIO_PATH}/mixin_mixout/mixin_mixout.c
-		${SOF_AUDIO_PATH}/mixin_mixout/mixin_mixout_generic.c
-		${SOF_AUDIO_PATH}/mixin_mixout/mixin_mixout_hifi3.c
-	)
-endif()
+zephyr_library_sources_ifdef(CONFIG_COMP_MIXER
+	${SOF_AUDIO_PATH}/mixer/mixer.c
+	${SOF_AUDIO_PATH}/mixer/mixer_generic.c
+	${SOF_AUDIO_PATH}/mixer/mixer_hifi3.c
+)
+
+zephyr_library_sources_ifdef(CONFIG_COMP_MIXIN_MIXOUT
+	${SOF_AUDIO_PATH}/mixin_mixout/mixin_mixout.c
+	${SOF_AUDIO_PATH}/mixin_mixout/mixin_mixout_generic.c
+	${SOF_AUDIO_PATH}/mixin_mixout/mixin_mixout_hifi3.c
```

Comment:
This makes the conditionals totally different from `src/audio/CMakeLists.txt` ?!?

Reply:
we use mmixin_mixout code to achieve the mixing function for IPC4 version and mixer code to achieve the mixing function for IPC3, but the infrastructure and implementation and even ability are very different for those 2 components.

> Either the IPC3 "mixer" and the IPC4 "mixin_mixout" are significantly different features that need a different name. Then don't add a new IPC4 COMP_MIXER. Instead a new, IPC4 COMP_MIXIN_MIXOUT option. Keep the single IPC3 COMP_MIXER.

so I think we should better use 2 different kconfig name with the dependence of IPC version for them, and since I move the IPC version dependence into kconfig, we have to change the src/audio/CMakeLists.txt to fit the kconfig modification.

Reply:
Great explanations, thank you! They _really_ deserve to be in the commit message.

> and since I move the IPC version dependence into kconfig, we have to change the src/audio/CMakeLists.txt to fit the kconfig modification.

... and as usual in these situations, the problem is: how are you going to TEST all these CMake changes? This requires compiling in many different ways.

Always remember: break it before you fix it. It's very common to edit some of our CMake files while accidentally NOT testing the changes because they're in a different use case or configuration than the current one. If you break it and nothing happens, then you're not testing it.

Reply:
Here's another very useful test trick in C files:

```C
#warning this volume Hifi7 file is being compiled
```

`#error` works too.

Reply:
actually even with the copy/past error, we won't found that error because the COMP_MIXER is default yes, we will always compile the mixin_mixout or mixer code. I noticed this because I want to change this to new hifi config style.